languages/common-lisp/exercises/concept/sameness/.meta/example.lisp

```diff
@@ -0,0 +1,13 @@
+(defpackage sameness
+  (:use :cl)
+  (:export :robot))
+
+(in-package :sameness)
+
+(defun robot (maze-id)
+  "Return a key to use for the doors in the maze designated by MAZE-ID."
+  (case maze-id
+    (:maze-1 #'eq)
+    (:maze-2 #'eql)
+    (:maze-3 #'eql)
+    (t (constantly nil))))
```

Comment:
If this returns `nil` then either the test code needs to handle it or it will signal an error (no function or some such) and that will drop the student into the debugger. Perhaps not what we'd want.

Which makes me wonder... does the test runner guard against the debugger being entered?

Reply:
Yeah the test-runner doesn't let the debugger trigger, so it would just report no function named nil and return that error to the user. With that being said, I think that `(constantly nil)` is a bit more robust and probably a less confusing error to beginners.